RestaurantManagementSystem/RestaurantManagementSystem/src/UseCases/Email Confirmation/EmailConfirmationHandler.cs

```diff
@@ -6,33 +6,28 @@ namespace RestaurantManagementSystem.src.UseCases.Email_Confirmation
 {
     public class EmailConfirmationHandler : IEmailConfirmationHandler
     {
-        private readonly IRestaurantRepo _restaurantRepo;
+        private readonly IRegistrationRequestRepo _restaurantRepo;
         private readonly IEmailService _emailservice;
-        public EmailConfirmationHandler (IRestaurantRepo restaurantRepo,IEmailService emailservice)
+        public EmailConfirmationHandler (IRegistrationRequestRepo restaurantRepo,IEmailService emailservice)
         {
             _restaurantRepo = restaurantRepo;
             _emailservice = emailservice;
         }
-        public async Task<bool> Confirm(ConfirmationCommand confirmationCommand)
+        public async Task<bool> VerifyEmail(ConfirmationCommand confirmationCommand)
         {
-            Guid id = new Guid(confirmationCommand.RequestId);
-            RegistrationRequest obj = await _restaurantRepo.FindByIdAsync(id);
-            if (obj.Status == RequestStatus.Status.PendingVerification)
-            {
-                RegistrationRequest Updated = RegistrationRequest.VertifyCode(confirmationCommand.ConfirmationCode, obj);
-                if (Updated.Status == RequestStatus.Status.WaitingForAdminResponse)
+            RegistrationRequest obj = await _restaurantRepo.FindByIdAsync(confirmationCommand.RequestId);
+            obj.VertifyCode(confirmationCommand.VerificationCode);
+            if (obj != null && obj.Status == Status.WaitingForAdminResponse)
```

Comment:
This layer should not check the request status and perform conditionally. you should throw errors as I mentioned above

Reply:
but it needs to check to update db if everything goes well

Reply:
if `VerifyCode` threw an error you will not update the DB.

Reply:
you don't have a catch here so it will exit the function and goes back to the controller and there is also no catch so the program might crash 😄